woocommerce/woocommerce-android
# [Tooling] Fix hotfix lanes

Part of the paaHJt-1WQ-p2 project.

## To test

 - Checkout this branch, run `bundle install`
 - Run `bundle exec fastlane new_hotfix_release version_name:7.0.1 version_code:231`
 - Verify that it cut a `release/7.0.1` branch from the `7.0` tag
 - Verify that it bumped the versionName and versionCode in `WooCommerce/build.gradle`
 - cherry-pick all the commits from this PR on top of the `release/7.0.1` branch. This is because we are gonna test the updated finalize lanes, and need to be on the hotfix release branch to test them. (You might get conflicts during the cherry-pick)
 - [Open the CircleCI page for WCAndroid](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-android) in advance, to be ready to cancel the release build
 - Run `bundle exec fastlane finalize_release`. You should get an error telling you that you should use `finalize_hotfix_release` for hotfix branches instead
 - Run `bundle exec fastlane finalize_hotfix_release` and verify that it triggered a release build, and that is is building it from the release/7.0.1 hotfix branch. Cancel the CI build immediately (NB: `231` is a version code that already exists so in case you fail to cancel it, the build should fail to upload to playstore anyway)
 - Delete the `release/7.0.1` branch (from both local and remote)

## PR Dependency ("status: do not merge" label)

This PR depends on https://github.com/wordpress-mobile/release-toolkit/pull/288 ; once this gets merged, we need to do a new release of the toolkit and update the reference in `Pluginfile` before merging this.
